main.py

```diff
@@ -10,43 +10,58 @@
-        contributors: Dict[str, List[Dict[str, str]]] = {
-            "Bloom Studio": data["servers"]["Bloom Studio"]["contributors"],
-            "Bloom Collective": data["servers"]["Bloom Collective"]["contributors"],
-        }
-        emoji_dicts: Dict[str, Dict[str, str]] = {
-            "Bloom Studio": data["servers"]["Bloom Studio"]["emoji_dictionary"],
-            "Bloom Collective": data["servers"]["Bloom Collective"]["emoji_dictionary"],
-        }
-    # Discord Config
-    intents: discord.Intents = discord.Intents.default()
-    intents.message_content = True
-    intents.reactions = True
-    bot: commands.Bot = commands.Bot(command_prefix="$", intents=intents)
-
-    # Setup the governance discord commands, and events
-    setup_gov_commands(bot)
-
-    # Setup the emotes discord commands, and events
-    setup_contrbitutor_commands(bot, contributors, emoji_dicts)
-    # Setup the shared events
-    setup_shared_events(bot, data, proposals, new_proposal_emoji)
-    bot.posted_events = load_posted_events()
-
-    # Setup the event discord commands, and events
-    setup_event_commands(bot)
-    setup_event_events(bot, emoji_dicts)
-
-    # Run the bot
-    bot.run(os.getenv("DISCORD_BOT_TOKEN"))
+
+
+class Bot:
+    def __init__(self):
+        self.bot = None
+        self.contributors = None
+        self.emoji_dicts = None
+        self.data = None
+
+    def load_data(self):
+        with open(CONTRIBUTORS_FILE_PATH, "r") as json_file:
+            self.data = json.load(json_file)
+            self.contributors = {
+                "Bloom Studio": self.data["servers"]["Bloom Studio"]["contributors"],
+                "Bloom Collective": self.data["servers"]["Bloom Collective"][
+                    "contributors"
+                ],
+            }
+            self.emoji_dicts = {
+                "Bloom Studio": self.data["servers"]["Bloom Studio"][
+                    "emoji_dictionary"
+                ],
+                "Bloom Collective": self.data["servers"]["Bloom Collective"][
+                    "emoji_dictionary"
+                ],
+            }
+
+    def setup_bot(self):
+        intents = discord.Intents.default()
+        intents.message_content = True
+        intents.reactions = True
+        intents.members = True
+        self.bot = commands.Bot(command_prefix="!", intents=intents)
+
+    def setup_commands_and_events(self):
+        setup_gov_commands(self.bot)
+        setup_contrbitutor_commands(self.bot, self.contributors, self.emoji_dicts)
+        setup_shared_events(self.bot, self.data, proposals, new_proposal_emoji)
+        self.bot.posted_events = load_posted_events()
+        setup_event_commands(self.bot)
+        setup_event_events(self.bot)
+
+    def run(self):
+        self.bot.run(os.getenv("DISCORD_BOT_TOKEN"))
+
+    def main(self):
+        self.load_data()
+        self.setup_bot()
+        self.setup_commands_and_events()
+        self.run()
```

Comment:
Purpose?

Reply:
> Purpose?

Not sure I follow exactly, can we hop on VC to discuss this point?

Reply:
> > Purpose?
> 
> Not sure I follow exactly, can we hop on VC to discuss this point?

Disregard, read through this again. It was showing me a LOT more lines when I first looked at this in the morning.

I used main to put everything together (kind of ties in with the logic behind your other comment asking of the purpose) instead of calling each function using the instance runner.

Reply:
> tbh bit exotic and convoluted
> 
> you now have multiple entry points so to say and well convoluting areas

Have simplified things in main.py :)

Reply:
> tbh bit exotic and convoluted
> 
> you now have multiple entry points so to say and well convoluting areas

This was actioned in another response to one of your comments.